src/libraries/System.Security.Cryptography/tests/ShakeTestDriver.cs

```diff
@@ -0,0 +1,448 @@
+                    Assert.Equal(kat.Output, Convert.ToHexString(hash), ignoreCase: true);
+                }
+            }
+        }
+
+        [ConditionalFact(nameof(IsSupported))]
+        public async Task KnownAnswerTests_OneShot_HashDataAsync_Stream_Destination()
+        {
+            foreach ((string Msg, string Output) kat in Fips202Kats)
+            {
+                byte[] hash = new byte[kat.Output.Length / 2];
+
+                using (MemoryStream message = new MemoryStream(Convert.FromHexString(kat.Msg)))
+                {
+                    await TShakeTrait.HashDataAsync(message, hash);
+                    Assert.Equal(kat.Output, Convert.ToHexString(hash), ignoreCase: true);
+                }
+            }
+        }
+
+        [ConditionalFact(nameof(IsSupported))]
+        public void ArgValidation_OneShot_HashData_OutputLengthNegative()
+        {
+            byte[] source = new byte[1];
+
+            AssertExtensions.Throws<ArgumentOutOfRangeException>(
+                "outputLength",
+                () => TShakeTrait.HashData(source, outputLength: -1));
+
+            AssertExtensions.Throws<ArgumentOutOfRangeException>(
+                "outputLength",
+                () => TShakeTrait.HashData(new ReadOnlySpan<byte>(source), outputLength: -1));
+
+            AssertExtensions.Throws<ArgumentOutOfRangeException>(
+                "outputLength",
+                () => TShakeTrait.HashData(Stream.Null, outputLength: -1));
+
+            // This assert is not async - argument validation should occur synchronously.
+            AssertExtensions.Throws<ArgumentOutOfRangeException>(
+                "outputLength",
+                () => TShakeTrait.HashDataAsync(Stream.Null, outputLength: -1));
+        }
+
+        [ConditionalFact(nameof(IsSupported))]
+        public void ArgValidation_OneShot_HashData_StreamNotReadable()
+        {
+            byte[] buffer = new byte[1];
+
+            AssertExtensions.Throws<ArgumentException>(
+                "source",
+                () => TShakeTrait.HashData(UntouchableStream.Instance, buffer));
+
+            AssertExtensions.Throws<ArgumentException>(
+                "source",
+                () => TShakeTrait.HashDataAsync(UntouchableStream.Instance, buffer));
+
+            AssertExtensions.Throws<ArgumentException>(
+                "source",
+                () => TShakeTrait.HashData(UntouchableStream.Instance, outputLength: 1));
+
+            AssertExtensions.Throws<ArgumentException>(
+                "source",
+                () => TShakeTrait.HashDataAsync(UntouchableStream.Instance, outputLength: 1));
+        }
+
+        [ConditionalFact(nameof(IsSupported))]
+        public async Task ArgValidation_OneShot_HashDataAsync_Cancelled()
+        {
+            byte[] buffer = new byte[1];
+            CancellationToken cancelledToken = new CancellationToken(canceled: true);
+
+            await Assert.ThrowsAsync<TaskCanceledException>(
```

Comment:
@stephentoub Do you care if this is `ThrowsAsync<TaskCanceledException>` vs `ThrowsAnyAsync<OperationCanceledException>`?  I know you discourage people from catching TCE in favor of OCE, but don't remember if that caring extends to tests.

Reply:
If I had a time machine... alas.

If we do `ThrowsAsync<TaskCanceledException>`, it's because we want to know if we happen to break something, but we should be free to change any such throws to be for the base exception in the future.

Personally, I'd just do `ThrowsAnyAsync<OperationCanceledException>`.

Regardless, we should make sure any documentation we might create that documents exception types calls out `OperationCanceledException` rather than `TaskCanceledException`.